src/test/java/com/redhat/darcy/webdriver/TakeScreenshotTest.java

```diff
@@ -1,49 +1,74 @@
 package com.redhat.darcy.webdriver;
 
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import com.redhat.darcy.ui.DarcyException;
 import com.redhat.darcy.ui.FindableNotPresentException;
 import com.redhat.darcy.web.api.Browser;
 import com.redhat.darcy.webdriver.internal.TargetedWebDriver;
 import com.redhat.darcy.webdriver.testing.doubles.StubWebDriverElementContext;
 import com.redhat.darcy.webdriver.testing.doubles.StubWebDriverParentContext;
 
-import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.runners.MockitoJUnitRunner;
+import org.junit.runners.JUnit4;
 import org.openqa.selenium.NoSuchWindowException;
 import org.openqa.selenium.OutputType;
 
-@RunWith(MockitoJUnitRunner.class)
-public class TakeScreenshotTest {
-    @Mock
-    TargetedWebDriver mockedDriver;
-
-    Browser browser;
-
-    @Before
-    public void setupBrowser() {
-        if (browser == null) {
-            browser = new WebDriverBrowser(mockedDriver,
-                    new StubWebDriverParentContext(),
-                    new StubWebDriverElementContext());
-        }
-    }
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
 
+@RunWith(JUnit4.class)
+public class TakeScreenshotTest {
     @Test
-    public void shouldTakeScreenshot() {
-        browser.takeScreenshot();
-        verify(mockedDriver).getScreenshotAs(OutputType.FILE);
+    public void shouldTakeScreenshotAndWriteToOutputStream() throws IOException {
+        TargetedWebDriver mockedDriver = mock(TargetedWebDriver.class);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Browser browser = new WebDriverBrowser(mockedDriver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        byte[] data = new byte[] { 1, 2, 3 };
+
+        when(mockedDriver.getScreenshotAs(OutputType.BYTES))
+                .thenReturn(data);
+
+        browser.takeScreenshot(baos);
+
+        verify(mockedDriver).getScreenshotAs(OutputType.BYTES);
+        assertThat(baos.toByteArray(), equalTo(data));
```

Comment:
I would remove the verify check here; you're already testing that's called by virtue using the bytes it returns. However, we should verify the output stream is flushed and closed also. Not sure if the BAOS requires that to get the bytes out of it. If it does, then that's already being tested. If not, we'll have to test that explicitly.